src/org/traccar/api/resource/AttributeAliasResource.java

```
@@ -0,0 +1,93 @@
+            }
+            return Context.getAliasesManager().getDeviceAttributeAliases(deviceId);
+        } else {
+            return Context.getAliasesManager().getUserDevicesAttributeAliases(getUserId());
+        }
+    }
+
+    @POST
+    public Response add(AttributeAlias entity) throws SQLException {
+        Context.getPermissionsManager().checkReadonly(getUserId());
+        if (!Context.getPermissionsManager().isAdmin(getUserId())) {
+            Context.getPermissionsManager().checkDevice(getUserId(), entity.getDeviceId());
+        }
+        Context.getAliasesManager().addAttributeAlias(entity);
+        return Response.ok(entity).build();
+    }
+
+    @Path("{id}")
+    @PUT
+    public Response update(@PathParam("id") long id, AttributeAlias entity) throws SQLException {
+        Context.getPermissionsManager().checkReadonly(getUserId());
+        if (!Context.getPermissionsManager().isAdmin(getUserId())) {
+            AttributeAlias oldAttrbuteAlias = Context.getAliasesManager().getAttributeAliasById(entity.getId());
+            if (oldAttrbuteAlias != null && oldAttrbuteAlias.getDeviceId() != entity.getDeviceId()) {
+                Context.getPermissionsManager().checkDevice(getUserId(), oldAttrbuteAlias.getDeviceId());
+            }
+            Context.getPermissionsManager().checkDevice(getUserId(), entity.getDeviceId());
+        }
+        Context.getAliasesManager().updateAttributeAlias(entity);
+        return Response.ok(entity).build();
+    }
+
+    @Path("{id}")
+    @DELETE
+    public Response remove(@PathParam("id") long id) throws SQLException {
+        Context.getPermissionsManager().checkReadonly(getUserId());
+        if (!Context.getPermissionsManager().isAdmin(getUserId())) {
+            AttributeAlias attrbuteAlias = Context.getAliasesManager().getAttributeAliasById(id);
+            Context.getPermissionsManager().checkDevice(getUserId(),
+                    attrbuteAlias != null ? attrbuteAlias.getDeviceId() : 0);
```

I think we don't need null check here. Just let it crash.